### PR TITLE
Introduce template query Id

### PR DESCRIPTION
**Pull request by YannanGao-gs**
#### What type of PR is this?

Improvement

#### What does this PR do / why is it needed ?

Introduce template query Id

#### Which issue(s) this PR fixes:

Fixes #

### CODE REVIEW

File: legend-engine-xts-data-space/legend-engine-xt-data-space-generation/src/main/java/org/finos/legend/engine/generation/analytics/DataSpaceAnalyticsHelper.java [hunk 1]
```diff
@@ -398,9 +398,14 @@ else if (elementDoc instanceof Root_meta_pure_metamodel_dataSpace_analytics_Data
                     executableAnalysisResult.title = executable._title();
                     executableAnalysisResult.description = executable._description();
                     DataSpaceTemplateExecutableInfo templateExecutableInfo = new DataSpaceTemplateExecutableInfo();
+                    templateExecutableInfo.id = ((Root_meta_pure_metamodel_dataSpace_DataSpaceTemplateExecutable) executable)._id();
 
                     // get V1 lambda
-                    DataSpaceTemplateExecutable executableV1 = (DataSpaceTemplateExecutable) dataSpaceProtocol.executables.stream().filter(e ->  e instanceof DataSpaceTemplateExecutable && e.title.equals(executable._title()) && e.description.equals(executable._description())).findFirst().get();
+                    DataSpaceTemplateExecutable executableV1 = (DataSpaceTemplateExecutable) dataSpaceProtocol.executables.stream().filter(e ->
```
Comment (MauricioUyaguari) [inline legend-engine-xts-data-space/legend-engine-xt-data-space-generation/src/main/java/org/finos/legend/engine/generation/analytics/DataSpaceAnalyticsHelper.java:404]: we can just use the ID now to find the corresponding template executable now right?

File: legend-engine-xts-data-space/legend-engine-xt-data-space-generation/src/main/java/org/finos/legend/engine/generation/analytics/DataSpaceAnalyticsHelper.java [hunk 2]
```diff
@@ -398,9 +398,14 @@ else if (elementDoc instanceof Root_meta_pure_metamodel_dataSpace_analytics_Data
...
+                            e.description == null ?
+                                    e instanceof DataSpaceTemplateExecutable && e.title.equals(executable._title()) :
```
Comment (MauricioUyaguari) [inline legend-engine-xts-data-space/legend-engine-xt-data-space-generation/src/main/java/org/finos/legend/engine/generation/analytics/DataSpaceAnalyticsHelper.java:406]: we do have tests that test this right?

**Review by MauricioUyaguari — CHANGES_REQUESTED** [review]

File: legend-engine-xts-data-space/legend-engine-xt-data-space-grammar/src/main/antlr4/org/finos/legend/engine/language/pure/grammar/from/antlr4/DataSpaceParserGrammar.g4 [hunk 3]
```diff
@@ -155,6 +156,8 @@ executablePath:                     DATA_SPACE_EXECUTABLE COLON qualifiedName SE
 ;
 executableTemplateQuery:            DATA_SPACE__TEMPLATE_QUERY COLON combinedExpression SEMI_COLON
 ;
+executableTemplateQueryId:          DATA_SPACE__TEMPLATE_QUERY__ID COLON (VALID_STRING | INTEGER | DECIMAL) SEMI_COLON
```
Comment (YannanGao-gs) [inline legend-engine-xts-data-space/legend-engine-xt-data-space-grammar/src/main/antlr4/org/finos/legend/engine/language/pure/grammar/from/antlr4/DataSpaceParserGrammar.g4:159]: Just a heads-up
123 will be prioritized as INTEGER 
2d will be treated as DECIMAL 
even if we define fragment ValidString  as  (Letter | Digit | '_' ) (Letter | Digit | '_' | '$')*
to allow pure digits and or some digit + 'd' 
we have to specify INTEGER | DECIMAL

Comment (YannanGao-gs) [reply legend-engine-xts-data-space/legend-engine-xt-data-space-grammar/src/main/antlr4/org/finos/legend/engine/language/pure/grammar/from/antlr4/DataSpaceParserGrammar.g4:159]: Besides, ID is designed to not have quotes.  Pure digits are allowed e.g. id: 123
only string follows this pattern, (Letter | Digit | '' ) (Letter | Digit | '' | '$')*, will be allowed, which means id shouldn't contain white space 


a heads-up, as I pointed it out above, digits + 'd'/'D' will be prioritized as decimal.

**Pull request merged**